packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/analytics_client.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/event_client.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/dart_queued_item_store.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/queued_item_store.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/flutter_provider_interfaces/device_context_info_provider.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/src/pinpoint/pinpoint_client.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
+import 'package:meta/meta.dart';
+
+/// {@template amplify_analytics_pinpoint.analytics_client}
+/// A builder class that constructs and exposes the [eventClient] and [endpointClient] for interacting with Pinpoint.
+/// {@endtemplate}
+class AnalyticsClient {
+  /// {@macro amplify_analytics_pinpoint.analytics_client}
+  AnalyticsClient({
+    DeviceContextInfoProvider? deviceContextInfoProvider,
+    EndpointInfoStoreManager? endpointInfoStoreManager,
+  })  : _deviceContextInfoProvider = deviceContextInfoProvider,
+        _endpointInfoStoreManager = endpointInfoStoreManager ??
+            EndpointInfoStoreManager(
+              store: AmplifySecureStorageWorker(
+                config: AmplifySecureStorageConfig(
+                  scope: EndpointStorageScope.analyticsPinpoint.name,
+                ),
+              ),
+            );
+
+  final DeviceContextInfoProvider? _deviceContextInfoProvider;
+  final EndpointInfoStoreManager _endpointInfoStoreManager;
+
+  /// Allows mocking of [PinpointClient] for unit tests
+  @visibleForTesting
+  Future<void> initWithClient({
+    required String pinpointAppId,
+    required PinpointClient pinpointClient,
+    QueuedItemStore? eventStore,
+  }) async {
+    final deviceContextInfo =
+        await _deviceContextInfoProvider?.getDeviceInfoDetails();
+
+    await _endpointInfoStoreManager.init(pinpointAppId: pinpointAppId);
+
+    endpointClient = EndpointClient(
+      pinpointAppId: pinpointAppId,
+      pinpointClient: pinpointClient,
+      endpointInfoStoreManager: _endpointInfoStoreManager,
+      deviceContextInfo: deviceContextInfo,
+    );
+
+    eventClient = EventClient(
+      pinpointAppId: pinpointAppId,
+      pinpointClient: pinpointClient,
+      endpointClient: endpointClient,
+      eventStore: eventStore,
+      deviceContextInfo: deviceContextInfo,
+    );
+  }
+
+  /// Async initialize the [eventClient] and [endpointClient].
+  /// Requires missing values typically provided in plugin configure.
+  Future<void> init({
+    required String pinpointAppId,
+    required String region,
+    required AWSCredentialsProvider authProvider,
+    DartQueuedItemStore? eventStore,
+  }) async {
+    final pinpointClient = PinpointClient(
```

Comment:
nit: If this could be injected, then we don't need to add `initWithClient`?

Reply:
`initWithClient` is specifically added for testing, see the comments in the method declaration: 

```
/// Allows mocking of [PinpointClient] for unit tests
  @visibleForTesting
  Future<void> initWithClient({
```

Reply:
Hui's asking why this can't be done in the constructor. And I agree - injecting dependencies in both the constructor and `init` is not a common pattern and is rather confusing.

Reply:
Thanks for clarifying.  I agree, it would be ideal to pass PinpointClient in constructor. 

Issue is that AnalyticsClient is a DI which means it must be created in the Flutter top level. 

And we only receive appId, region, credentialsProvider in the configure method.  

Furthermore we shouldn't only have one `init` method with PinpointClient because `AnalyticsClient` is used by PushNotifications which must avoid a dependency on the Pinpoint SDK.